include/swift/Runtime/Win32.h

```diff
@@ -0,0 +1,55 @@
+//===--- Win32.h - Win32 utility functions ----------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Utility functions that are specific to the Windows port.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_RUNTIME_WIN32_H
+#define SWIFT_RUNTIME_WIN32_H
+
+#ifdef _WIN32
+
+#include "swift/shims/Visibility.h"
+
+#include <wchar.h>
+
+namespace swift {
+namespace win32 {
+
+/// Convert a wide string to UTF-8.
+///
+/// @param str The string to convert.
+///
+/// @returns The string, converted to UTF-8.  The caller is responsible
+///          for freeing this string with @c free() when done with it.
+///
+/// If @a str cannot be converted to UTF-8, @c nullptr is returned.
+SWIFT_RUNTIME_STDLIB_INTERNAL
+char *copyUTF8FromWide(const wchar_t *str);
```

Comment:
Elsewhere where we have Win32-specific symbols, I used `[_]swift_win32_` as a prefix. Although these symbols aren't going to be exported, you might want to adopt that prefix for consistency instead of using a `win32` nested namespace.

Reply:
Since these are C++ symbols, I'm not sure why we would want to prefer C-style prefixes over C++-style namespaces. If we're exporting a C API, sure, putting `[_]swift_win32_` on the front would be the right choice.

Reply:
In fact, we can't put them in a C++ namespace because `SWIFT_RUNTIME_STDLIB_INTERNAL` does `extern "C"`. So yes, we'll rename them, as you suggest.